pkg/storage/etcd/etcd_watcher.go

```diff
@@ -359,9 +359,6 @@ func (w *etcdWatcher) sendAdd(res *etcd.Response) {
 		return
 	}
 	action := watch.Added
-	if res.Node.ModifiedIndex != res.Node.CreatedIndex {
```

Comment:
hmm, I guess routing EtcdGet to sendAdd was already implying that was the only code path that could get here. I think you're right that that is the only way it can reach that point today, but that feels like baking assumptions into the response handling